app/uk/gov/hmrc/gform/sdes/SdesService.scala

```diff
@@ -139,15 +148,44 @@ class SdesService(
     showBeforeAt: Option[Boolean],
     destination: Option[SdesDestination]
   ): Future[SdesSubmissionPageData] = {
+    val skip = page * pageSize
+    doSearch(Some((skip, pageSize)), processed, formTemplateId, status, showBeforeAt, destination, None)
+  }
+
+  override def searchAll(
+    processed: Option[Boolean],
+    formTemplateId: Option[FormTemplateId],
+    status: Option[NotificationStatus],
+    showBeforeAt: Option[Boolean],
+    destination: Option[SdesDestination],
+    showBeforeLastUpdatedAt: Option[Int]
+  ): Future[SdesSubmissionPageData] =
+    doSearch(None, processed, formTemplateId, status, showBeforeAt, destination, showBeforeLastUpdatedAt)
+
+  private def doSearch(
+    maybeSkipAndPageSize: Option[(Int, Int)],
+    processed: Option[Boolean],
+    formTemplateId: Option[FormTemplateId],
+    status: Option[NotificationStatus],
+    showBeforeAt: Option[Boolean],
+    destination: Option[SdesDestination],
+    showBeforeLastUpdatedAt: Option[Int]
+  ): Future[SdesSubmissionPageData] = {
 
     val queryByTemplateId =
       formTemplateId.fold(exists("_id"))(t => equal("formTemplateId", t.value))
     val queryByProcessed =
       processed.fold(queryByTemplateId)(p => Filters.and(equal("isProcessed", p), queryByTemplateId))
     val queryByStatus =
       status.fold(queryByProcessed)(s => Filters.and(equal("status", fromName(s)), queryByProcessed))
+
+    val queryByLastUpdated =
+      showBeforeLastUpdatedAt.fold(queryByStatus)(d =>
+        Filters.and(lt("lastUpdated", LocalDateTime.now().minusHours(d.toLong)), queryByStatus)
```

Comment:
Are you sure you want to query by `lastUpdated`?

Reply:
Not 100% sure. The other option is  `submittedAt`.  My reasoning was I that I don't want to touch the submissions that's being changes in the last 7 hours.

Reply:
What do you think?

Reply:
`lastUpdated` will be missing for `SdesSubmission` you care about. You can use `submittedAt` or `createdAt` depending on what semantics are you implementing.

Reply:
`And the submissions were done prior to midnight` is in AC.  Clearly  that indicates that  `submittedAt` should be used. I was wrong. will make changes.